### PR TITLE
Make the DWARF version configurable.

### PR DESCRIPTION
**Pull request by adrian-prantl**
This mirrors a similar change in the swift repository.

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test with https://github.com/apple/swift/pull/69254

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/69254
@swift-ci test

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -2308,7 +2308,24 @@ extension Driver {
       diagnosticsEngine.emit(.error_argument_not_allowed_with(arg: fullNotAllowedOption, other: levelOption.spelling))
     }
 
-    return DebugInfo(format: format, level: level, shouldVerify: shouldVerify)
+    // Determine the DWARF version.
+    var dwarfVersion: UInt8 = 4
+    if let versionArg = parsedOptions.getLastArgument(.dwarfVersion) {
+      let parsedVersion = UInt8(versionArg.asSingle)
+      if parsedVersion != nil && parsedVersion! >= 2 && parsedVersion! <= 5 {
+        dwarfVersion = parsedVersion!
```
Comment (DougGregor) [inline Sources/SwiftDriver/Driver/Driver.swift:2316]:
```suggestion
      let parsedVersion = UInt8(versionArg.asSingle)
      if let parsedVersion = UInt8(versionArg.asSingle), parsedVersion >= 2 && parsedVersion <= 5 {
        dwarfVersion = parsedVersion
```

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 2]
```diff
@@ -2308,7 +2308,24 @@ extension Driver {
       diagnosticsEngine.emit(.error_argument_not_allowed_with(arg: fullNotAllowedOption, other: levelOption.spelling))
     }
 
-    return DebugInfo(format: format, level: level, shouldVerify: shouldVerify)
+    // Determine the DWARF version.
+    var dwarfVersion: UInt8 = 4
```
Comment (DougGregor) [inline Sources/SwiftDriver/Driver/Driver.swift:2312]: Usually we just use `Int`, but this is okay.

**Review by DougGregor — APPROVED** [review]
One suggested cleanup to remove the unsightly `!`s, but otherwise this LGTM

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/69254
@swift-ci test

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/69254
@swift-ci test windows

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/69254
@swift-ci test windows

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/69254
@swift-ci test

**Review by artemcm — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Pull request merged**